src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -353,31 +372,83 @@ internal void UpdateWithExemplar(long number, ReadOnlySpan<KeyValuePair<string,
             {
                 case AggregationType.LongSumIncomingDelta:
                     {
-                        Interlocked.Add(ref this.runningValue.AsLong, number);
+                        var sw = default(SpinWait);
+                        while (true)
+                        {
+                            if (Interlocked.Exchange(ref this.mpComponents.IsCriticalSectionOccupied, 1) == 0)
+                            {
+                                // Lock acquired
+                                unchecked
+                                {
+                                    this.runningValue.AsLong += number;
```

Comment:
This is not thread-safe. There might be another user thread trying to update `runningValue.AsLong` with `Interlocked.Increment` in the original `Update(long value)` method.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3a89cb38b0161183121ce973837c92a6f7f670c6/src/OpenTelemetry/Metrics/AggregatorStore.cs#L342-L350

Reply:
good catch. Will follow up